internal/direct_util.go

```diff
@@ -0,0 +1,51 @@
+// Package internal contains private helper functions needed in client and server
+package internal
+
+import (
+	"fmt"
+
+	"github.com/google/go-tpm/tpmutil"
+
+	"github.com/google/go-tpm/direct/structures/tpm"
+	"github.com/google/go-tpm/direct/structures/tpml"
+	"github.com/google/go-tpm/direct/structures/tpms"
+)
+
+// CreateTPMSPCRSelection will create a TPMS PCR Selection given a slice of uint32 and a tpm.AlgID.
+// This function only supports up to 24 PCRs.
+func CreateTPMSPCRSelection(s []uint32, hash tpm.AlgID) (*tpms.PCRSelection, error) {
```

Comment:
why return a pointer to `tpms.PCRSelection`?

Reply:
Resolved. Was following the tpmDirect format in go-tpm however it does not make sense for a higher level library to deal with unnecessary pointers.